### ISSUE TITLE
Add the ability to set margins

### ISSUE THREAD

**Issue by nicolasvac** [picqer/php-barcode-generator#123]
Hello everyone, we are actively using this library to generate barcodes, and we found out that some mobile devices have troubles reading it because there is no ability to set margins around the image, to give a bit of space before starting to read the lines.

While i know a simple solution like this can be applied:
```php
      $generatedImage = new Imagick();
      $generatedImage->readImageBlob($barcodeGenerator->getBarcode(xx));
      $generatedImage->borderImage('white', 20, 20);

      $barcode = $generatedImage->getImageBlob();
```

It would be nice to have the ability for both SVG, PNG and JPG to set the margins. 
Thank you.

**Comment by casperbakker** [comment]
Yes that would be nice. I think the current interface does not work for these kind of options. But I am exploring options to divide the barcode encoding and the image rendering parts. That would make it easier to differentiate the image renderers.

**Comment by osflake** [comment]
Any changes? Please @casperbakker some feedback
